pkg/storage/util/util_test.go

```diff
@@ -0,0 +1,70 @@
+package util
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+
+	configv1 "github.com/openshift/api/config/v1"
+	imageregistryv1 "github.com/openshift/cluster-image-registry-operator/pkg/apis/imageregistry/v1"
+	regopclient "github.com/openshift/cluster-image-registry-operator/pkg/client"
+)
+
+var (
+	infrastructureName = "jsmith-xhv4"
+	prefix             = fmt.Sprintf("%s-%s", infrastructureName, imageregistryv1.ImageRegistryName)
+)
+
+type MockInfrastructureLister struct {
+}
+
+func (m MockInfrastructureLister) Get(name string) (*configv1.Infrastructure, error) {
+	return &configv1.Infrastructure{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "cluster",
+		},
+		Status: configv1.InfrastructureStatus{
+			InfrastructureName: infrastructureName,
+			PlatformStatus:     &configv1.PlatformStatus{},
+		},
+	}, nil
+}
+
+func (m MockInfrastructureLister) List(selector labels.Selector) ([]*configv1.Infrastructure, error) {
+	var list []*configv1.Infrastructure
+	return list, nil
+}
+
+func TestGenerateStorageName(t *testing.T) {
+	l := regopclient.Listers{Infrastructures: MockInfrastructureLister{}}
+	tests := [][]string{
+		// test with no additionals
+		{},
```

Comment:
Maybe add an extra `nil` test case here? In practice the behavior will be the same as with `{}` but it may help in the future.

Reply:
Good idea, added in next push